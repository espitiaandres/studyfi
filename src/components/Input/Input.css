.form {
  border-top: 2px solid #D3D3D3;
  display: flex;
}
  
.input {
  border: none;
  border-radius: 0;
  font-size: 16px;
  padding: 2%;
  width: 100%;
}

input:focus, textarea:focus, select:focus{
  outline: none;
}

.sendButton {
  background: #1ED760;
  border: none;
  color: #fff !important;
  display: inline-block;
  text-transform: uppercase;
  text-decoration: none;
  padding: 10px;
  width: 20%;
}

.sendButton:hover {
  cursor: pointer;
}
