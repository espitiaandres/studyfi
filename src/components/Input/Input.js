//
//  Input.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import './Input.css';

const Input = ({ message, setMessage, sendMessage }) => (
    <form className="form">
        <input
            className="input"
            type="text"
            placeholder="Enter a message..."
            value={message} 
            onChange={(event) => setMessage(event.target.value)}
            onKeyPress={event => event.key === 'Enter' ? sendMessage(event) : null}
        />
        <button className="sendButton" onClick={(event) => sendMessage(event)}>Send</button>
    </form>
)

export default Input;
