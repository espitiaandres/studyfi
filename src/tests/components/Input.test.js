//
//  Input.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import Input from '../../components/Input/Input';

test('should render Input component correctly', () => {
    const wrapper = shallow(<Input />);
    expect(wrapper).toMatchSnapshot();
});
