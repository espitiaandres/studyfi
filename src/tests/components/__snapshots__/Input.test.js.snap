// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input component correctly 1`] = `ShallowWrapper {}`;
